clientv3/integration/metrics_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bufio"
+	"io"
+	"math/rand"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/pkg/testutil"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/net/context"
+)
+
+var (
+	host string = "localhost"
+	port string = "27989"
+)
+
+func TestV3ClientMetrics(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	var (
+		ln  net.Listener
+		err error
+	)
+
+	// listen for all prometheus metrics
+	go func() {
+		// listen for all prometheus metrics
+		go func() {
+			http.Handle("/metrics", prometheus.Handler())
+			srv := &http.Server{Addr: ":" + port, Handler: nil}
+			addr := srv.Addr
+			if addr == "" {
+				addr = ":http"
+			}
+			ln, err = net.Listen("tcp", addr)
+
+			if err != nil {
+				t.Errorf("Error : %v occured while listening on addr : %v", err, addr)
+			}
+
+			err = srv.Serve(ln.(*net.TCPListener))
+
+			if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
+				t.Errorf("Error occured while getting connection : %v\n", err)
+			}
+		}()
+	}()
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
```

Comment:
@xiang90 currently the integration framework is designed to have one client per member. Hence i had to make it 3.